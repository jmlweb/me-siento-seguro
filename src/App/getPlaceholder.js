import { getMonth } from 'date-fns';

const getPlaceholder = () => {
  const currentDate = new Date();
  const currentMonth = getMonth(currentDate);
  if (currentMonth === 11) {
    return 'christmas, New year eve, snow...';
  }
  if (currentMonth > 5 && currentMonth < 9) {
    return 'summer, party, pool, sea...';
  }
  if (currentMonth > 1 && currentMonth < 6) {
    return 'spring, flowers, picnic...';
  }
  if (currentMonth > 8 && currentMonth < 11) {
    return 'autumn, leafs, halloween, costume...';
  }
  return 'winter, cold, snow...';
};

export default getPlaceholder;
