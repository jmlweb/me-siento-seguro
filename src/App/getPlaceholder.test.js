import getPlaceholder, { PLACEHOLDERS } from './getPlaceholder';

describe('getPlaceholder', () => {
  test('christmas', () => {
    const selDate = new Date('2018-12-01');
    expect(getPlaceholder(selDate)).toBe(PLACEHOLDERS.CHRISTMAS);
  });
  test('summer', () => {
    const selDate = new Date('2018-07-01');
    expect(getPlaceholder(selDate)).toBe(PLACEHOLDERS.SUMMER);
  });
  test('summer', () => {
    const selDate = new Date('2018-03-01');
    expect(getPlaceholder(selDate)).toBe(PLACEHOLDERS.SPRING);
  });
  test('autumn', () => {
    const selDate = new Date('2018-10-01');
    expect(getPlaceholder(selDate)).toBe(PLACEHOLDERS.AUTUMN);
  });
  test('winter', () => {
    const selDate = new Date('2018-1-01');
    expect(getPlaceholder(selDate)).toBe(PLACEHOLDERS.WINTER);
  });
  test('snapshot', () => {
    const selDate = new Date('2018-12-01');
    expect(getPlaceholder(selDate)).toMatchSnapshot();
  });
});
