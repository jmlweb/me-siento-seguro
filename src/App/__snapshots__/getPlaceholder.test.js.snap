// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPlaceholder snapshot 1`] = `"christmas, New year eve, snow..."`;
